### PR TITLE
Make transports optional

### PR DESCRIPTION
**Pull request by dmitry-markin**
Put extra transports behind feature flags to make some dependencies optional.

While at it, also made sure no warnings are emitted on the recent rust version.

Resolves https://github.com/paritytech/litep2p/issues/161.

### CODE REVIEW

File: src/transport/manager/handle.rs [hunk 1]
```diff
@@ -285,7 +287,7 @@ pub struct TransportHandle {
     pub protocols: HashMap<ProtocolName, ProtocolContext>,
     pub next_connection_id: Arc<AtomicUsize>,
     pub next_substream_id: Arc<AtomicUsize>,
-    pub protocol_names: Vec<ProtocolName>,
+    pub _protocol_names: Vec<ProtocolName>,
```
Comment (lexnv) [inline src/transport/manager/handle.rs:290]: dq: Do we plan to use this in the future?

File: tests/connection/mod.rs [hunk 2]
```diff
@@ -21,30 +21,46 @@
...
     protocol::libp2p::ping::{Config as PingConfig, PingEvent},
-    transport::{
-        quic::config::Config as QuicConfig, tcp::config::Config as TcpConfig,
-        websocket::config::Config as WebSocketConfig,
-    },
+    transport::tcp::config::Config as TcpConfig,
     Litep2p, Litep2pEvent, PeerId,
 };
 
+#[cfg(feature = "websocket")]
```
Comment (lexnv) [inline tests/connection/mod.rs:30]: nit: It might be worth having this module structured as:
- `mod.rs` to bridge the `quic.rs` `websocket.rs` and `tcp.rs` depending on feature flags; and here we can have the common logic as well
- then have the tests moved to individual files so that the only `cfg` we state is in mod.rs


```
#mod.rs

#[cfg(feature = "quic")]
use quic.rs
...
```

We can also handle this later since I expect to run into some edge-cases along the way which may involve a closer refactor

File: tests/protocol/identify.rs [hunk 3]
```diff
@@ -26,19 +26,33 @@ use litep2p::{
...
 enum Transport {
-    Quic(QuicConfig),
     Tcp(TcpConfig),
+    #[cfg(feature = "quic")]
+    Quic(QuicConfig),
+    #[cfg(feature = "websocket")]
     WebSocket(WebSocketConfig),
 }
 
+fn add_transport(config: ConfigBuilder, transport: Transport) -> ConfigBuilder {
```
Comment (lexnv) [inline tests/protocol/identify.rs:46]: nit: By looking at this fn, I believe we can extract some common bits into tests/common.rs? Can be done as a followup :D

**Review by lexnv — APPROVED** [review]
LGTM! Nice work here 👍 

I like that we guard dependencies by feature flags. This should be quite important for users that only desire individual transports. It adds a bit of complexity for us in managing feature-flags, and we can look into the future at structuring our testing in a friendlier way for us, although put in balance I believe this is good 🙏

File: tests/connection/mod.rs (hunk 2 above)
Comment (dmitry-markin) [reply tests/connection/mod.rs:30]: I was thinking about putting quick/websocket/webrtc tests into dedicated files, but it looks like the tests only differ in the transport used, and belong better to the connection/protocol test files. I.e., it would be easier to have these tests along each other in the same file. I will think if there is another alternative to having `cfg` spread over the files. Let me know if you have anything in mind

File: tests/protocol/identify.rs (hunk 3 above)
Comment (dmitry-markin) [reply tests/protocol/identify.rs]: Done. While doing this I ran into issue that `tests/common.rs` is considered to be a test on it's own, and rust generates dead code warnings. So I ended up putting it in `tests/common/mod.rs`.

File: src/transport/manager/handle.rs (hunk 1 above)
Comment (dmitry-markin) [reply src/transport/manager/handle.rs]: I think we can safely remove it for now (done).

**Comment by nazar-pc** [comment]
Thank you, very appreciated!
